data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/activities/TemplateParamsUpgradePlugin.java

```diff
@@ -0,0 +1,118 @@
+import org.exoplatform.container.PortalContainer;
+import org.exoplatform.container.component.RequestLifeCycle;
+import org.exoplatform.container.xml.InitParams;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+public class TemplateParamsUpgradePlugin extends UpgradeProductPlugin {
+  private static final Log LOG = ExoLogger.getExoLogger(TemplateParamsUpgradePlugin.class);
+
+  private final PortalContainer container;
+
+  private final EntityManagerService entityManagerService;
+
+  private static final String  OLD_TEMPLATE_PARAMS_KEY = "old.template.params.key";
+
+  private static final String  NEW_TEMPLATE_PARAMS_KEY = "new.template.params.key";
+
+  private int                  templatePramasUpdatedCount;
+
+  private String                oldTemplateParamskey;
+
+  private String                newTemplateParamskey;
+
+  public TemplateParamsUpgradePlugin(PortalContainer container, EntityManagerService entityManagerService, InitParams initParams) {
+    super(initParams);
+    this.container = container;
+    this.entityManagerService = entityManagerService;
+
+    if (initParams.containsKey(OLD_TEMPLATE_PARAMS_KEY)) {
+      oldTemplateParamskey = initParams.getValueParam(OLD_TEMPLATE_PARAMS_KEY).getValue();
+    }
+    if (initParams.containsKey(NEW_TEMPLATE_PARAMS_KEY)) {
+      newTemplateParamskey = initParams.getValueParam(NEW_TEMPLATE_PARAMS_KEY).getValue();
+    }
+  }
+
+  @Override
+  public boolean shouldProceedToUpgrade(String newVersion,
+                                        String previousGroupVersion,
+                                        UpgradePluginExecutionContext previousUpgradePluginExecution) {
+    int executionCount = previousUpgradePluginExecution == null ? 0 : previousUpgradePluginExecution.getExecutionCount();
+    return !isExecuteOnlyOnce() || executionCount == 0;
+  }
+
+  @Override
+  public void processUpgrade(String oldVersion, String newVersion) {
+    if (StringUtils.isEmpty(oldTemplateParamskey)||StringUtils.isEmpty(newTemplateParamskey)) {
+      LOG.error("Couldn't process upgrade, all parameters are mandatory");
+      return;
+    }
+    long startupTime = System.currentTimeMillis();
+
+    ExoContainerContext.setCurrentContainer(container);
+    boolean transactionStarted = false;
+
+    LOG.info("Start upgrade of ACTIVITY_TEMPLATE_PARAMS with key {} to use {}", oldTemplateParamskey,newTemplateParamskey);
+    RequestLifeCycle.begin(this.entityManagerService);
+    EntityManager entityManager = this.entityManagerService.getEntityManager();
+    try {
+      if (!entityManager.getTransaction().isActive()) {
+        entityManager.getTransaction().begin();
+        transactionStarted = true;
+      }
+
+      String sqlString = "UPDATE SOC_ACTIVITY_TEMPLATE_PARAMS SET TEMPLATE_PARAM_KEY = TRIM(TEMPLATE_PARAM_KEY) WHERE TEMPLATE_PARAM_KEY LIKE '"+oldTemplateParamskey+"'"
```

Comment:
You missed to update the query 
There is no need for the check `"AND TEMPLATE_PARAM_KEY LIKE '%  '"` since we will put the value with spaces in the init params

Reply:
I think we need to keep this condition to check the space. I am trying with both the operators '=' and 'LIKE' that ignore any trailing spaces when comparing strings. WDYT ?

Reply:
@sofyenne IMO no need since it will limit the usage of the upgrade plugin.

Reply:
@ahamdi Query updated !